src/solvers/dg_curve_quad/curve_interpolant.jl

```diff
@@ -0,0 +1,122 @@
+
+mutable struct GammaCurve{RealT<:Real, NNODES}
+  nodes          ::SVector{NNODES, RealT}
+  bary_weights   ::SVector{NNODES, RealT}
+  x_vals         ::SVector{NNODES, RealT}
+  y_vals         ::SVector{NNODES, RealT}
+end
```

Comment:
Do you really want to have a `mutable struct` here or is a `struct` (immutable) also okay?